core/src/main/java/com/backblaze/b2/json/B2JsonObjectHandler.java

```diff
@@ -315,13 +356,15 @@ public void serialize(T obj, B2JsonWriter out) throws IOException, B2JsonExcepti
                         out.writeString(unionTypeFieldValue);
                         typeFieldDone = true;
                     }
-                    out.writeObjectFieldNameAndColon(fieldInfo.getName());
-                    final Object value = fieldInfo.field.get(obj);
-                    if (fieldInfo.requirement == FieldRequirement.REQUIRED && value == null) {
-                        throw new B2JsonException("required field " + fieldInfo.getName() + " cannot be null");
+                    if (fieldInfo.isInVersion(version)) {
+                        out.writeObjectFieldNameAndColon(fieldInfo.getName());
+                        final Object value = fieldInfo.field.get(obj);
+                        if (fieldInfo.isRequiredInVersion(version) && value == null) {
```

Comment:
the code here and the existence of isRequiredInVersion makes me think that i should be able to say "became option in version X" or "required starting in version Y", but i don't know how to do that. right now, i *think* requiredness is orthogonal to version info, so it's not yet possible to express those more complicated restrictions.  that's fine -- as long as we don't need them, simpler is better.  :)

i just want to check....am i missing something?

Reply:
maybe isRequiredInVersion() is really isRequiredAndInVersion() right now?  (it's implemented that way.)